File: C/5th Week/palindrome.c
// 30. Check if a number is a palindrome
#include <stdio.h>

void main()
{
    int n, tmp, rev = 0;
    printf("Enter a number: ");
    scanf("%i", &n);
    tmp = n;
    while (tmp != 0)
    {
        rev = rev * 10 + tmp % 10;
        tmp /= 10;
    }
    if (rev == n)
    {
        printf("%i is a palindrome.\n", n);
    }
    else
    {
        printf("%i is not a palindrome.\n", n);
    }
}

File: java Programming/1st Week/palindrome.java
// 8. Write a Program to find if the given number is palindrome or not.

import java.util.Scanner;

public class palindrome {
   public static void main(String[] args) {
      Scanner sc = new Scanner(System.in);
      System.out.print("Enter a number: ");
      int original = sc.nextInt();
      sc.close();
      int tmp = original, reverse = 0;
      while (tmp != 0) {
         reverse = reverse * 10 + tmp % 10;
         tmp /= 10;
      }
      if (original == reverse) {
         System.out.println(original + " is a palindrome.");
      } else {
         System.out.println(original + " is not a palindrome.");
      }
   }
}
